packages/components/pie-thumbnail/src/index.ts

```diff
@@ -27,21 +28,48 @@ export class PieThumbnail extends LitElement implements ThumbnailProps {
     @property({ type: String })
     public alt = defaultProps.alt;
 
+    @property({ type: Boolean })
+    public disabled = defaultProps.disabled;
+
+    @property({ type: Boolean })
+    public hasPadding = defaultProps.hasPadding;
+
+    @property({ type: String })
+    public backgroundColor = defaultProps.backgroundColor;
+
+    @property({ type: Object })
+    public placeholder: ThumbnailProps['placeholder'];
+
+    private handleImageError = () => {
+        if (this.placeholder?.src) this.src = this.placeholder.src as string;
+        if (this.placeholder?.alt) this.alt = this.placeholder.alt as string;
+    };
+
     render () {
         const {
             variant,
             src,
             alt,
+            disabled,
+            hasPadding,
+            backgroundColor,
+            handleImageError,
         } = this;
 
         const wrapperClasses = {
             'c-thumbnail': true,
             [`c-thumbnail--${variant}`]: true,
+            'c-thumbnail--disabled': disabled,
+            'c-thumbnail--padding': hasPadding,
+        };
+
+        const wrapperStyles = {
+            backgroundColor,
```

Comment:
would be interesting to know why `backgroundColor` is so open, rather than just having a set number of background colors to pick from? Can see this getting misused potentially if any colour can be specified.

Reply:
the background color of the thumbnail component can be any background color as specified by design specs.

Reply:
agreed

Reply:
could you please share how you think we should proceed with this? 
I don't think manually listing all background color tokens is maintainable.
this is the designs note

``
When using transparent PNGs the background colour is set to $container-default, and can be changed as need to any container colour token.
``

Reply:
we updated to follow the variants approach as discussed offline.